server/datastore/mysql/hosts.go

```diff
@@ -1926,68 +1943,83 @@ WHERE
 // Aggregated stats for os versions are stored by team id with 0 representing the global case
 // If existing team has no hosts, we explicity set the json value as an empty array
 func (ds *Datastore) UpdateOSVersions(ctx context.Context) error {
+	// TODO: counts will be lagging until offline hosts report back with detail query
+
 	sql := `
 INSERT INTO aggregated_stats (id, type, json_value)
```

Comment:
Is it just me or is this query is getting out of hands? :/ (should we start moving away from `aggregated_stats` in future features? or just consider splitting the query into two: read-then-write?)

Reply:
Totally agree!

Reply:
Hm, yeah, `INSERT ... FROM SELECT` can be dangerous in terms of locking and whatnot. How hard would it be to break this down to also avoid subqueries?

Reply:
+1. Worth giving it a shot (though we have two days before freeze).